test/distributed/fsdp/test_wrap.py

```diff
@@ -367,21 +370,26 @@ def test_always_wrap(self):
 
     @unittest.skipIf(torch.cuda.device_count() < 2, "Requires at least 2 GPUs")
     def test_transformer_auto_wrap_policy(self):
-        model = TransformerWithSharedParams(group=self.process_group)
-        my_auto_wrap_policy = functools.partial(
+        """Tests the ``transformer_auto_wrap_policy``."""
+        auto_wrap_policy = functools.partial(
             transformer_auto_wrap_policy,
-            transformer_layer_cls={TransformerEncoderLayer, TransformerDecoderLayer}
+            transformer_layer_cls={TransformerEncoderLayer, TransformerDecoderLayer},
         )
-        fsdp_model = FSDP(
-            model,
-            process_group=self.process_group,
-            auto_wrap_policy=my_auto_wrap_policy
+        fsdp_kwargs = {"auto_wrap_policy": auto_wrap_policy}
+        fsdp_model = TransformerWithSharedParams.init(
+            self.process_group,
+            FSDPInitMode.RECURSIVE,
+            CUDAInitMode.CUDA_BEFORE,
+            fsdp_kwargs,
         )
-        self.assertTrue(isinstance(fsdp_model, FSDP))
-        for layer in fsdp_model.module.transformer.encoder.layers:
-            self.assertTrue(isinstance(layer, FSDP))
-        for layer in fsdp_model.module.transformer.decoder.layers:
-            self.assertTrue(isinstance(layer, FSDP))
+        modules = list(fsdp_model.modules())
+        encoder_layers = set(fsdp_model.module.transformer.encoder.layers)
+        decoder_layers = set(fsdp_model.module.transformer.decoder.layers)
+        for module in modules:
+            if module is fsdp_model or module in encoder_layers or module in decoder_layers:
```

Comment:
maybe check `isinstance` to be more robust?

Reply:
I am not sure I follow the suggestion. `isinstance` is what we are checking below with the asserts, so it would not make sense to check it in the conditional.